src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -260,7 +263,10 @@ public void deleteRepository(VcsRepositoryUrl repositoryUrl) {
             gitlab.getProjectApi().deleteProject(repositoryId);
         }
         catch (GitLabApiException e) {
-            throw new GitLabException("Error trying to delete repository on GitLab: " + repositoryName, e);
+            // Do not throw an exception if we try to delete a non-existant repository.
+            if (e.getHttpStatus() != 404) {
```

Comment:
couldn't you use also the `HttpStatus.SC_NOT_FOUND` as in the deleteBuildPlan exception handling?